app/src/main/res/layout/taskdetail_frag_v2.xml

```diff
@@ -0,0 +1,90 @@
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:orientation="vertical"
+                android:paddingBottom="@dimen/activity_vertical_margin"
+                android:paddingLeft="@dimen/activity_horizontal_margin"
+                android:paddingRight="@dimen/activity_horizontal_margin"
+                android:paddingTop="@dimen/activity_vertical_margin"
+                android:visibility="@{viewmodel.isDataAvailable ? View.GONE : View.VISIBLE}">
+
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="No data"
+                    android:textAppearance="?android:attr/textAppearanceLarge" />
+            </LinearLayout>
+
+            <RelativeLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:orientation="vertical"
+                android:paddingBottom="@dimen/activity_vertical_margin"
+                android:paddingLeft="@dimen/activity_horizontal_margin"
+                android:paddingRight="@dimen/activity_horizontal_margin"
+                android:paddingTop="@dimen/activity_vertical_margin"
+                android:visibility="@{viewmodel.isDataAvailable ? View.VISIBLE : View.GONE}">
+
+                <CheckBox
+                    android:id="@+id/task_detail_complete"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center_vertical"
+                    android:layout_marginRight="@dimen/activity_horizontal_margin"
+                    android:onClick="@{(view) -> listener.onCompleteChanged(view)}"
+                    android:checked="@{viewmodel.completed}" />
+
+                <TextView
+                    android:id="@+id/task_detail_title"
+                    android:layout_toRightOf="@id/task_detail_complete"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:text="@{viewmodel.task.title}"
+                    android:textAppearance="?android:attr/textAppearanceLarge" />
+
+                <TextView
+                    android:id="@+id/task_detail_description"
+                    android:layout_toRightOf="@id/task_detail_complete"
+                    android:layout_below="@id/task_detail_title"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:text="@{viewmodel.task.description}"
+                    android:textAppearance="?android:attr/textAppearanceMedium" />
+
+            </RelativeLayout>
+        </LinearLayout>
+    </com.example.android.architecture.blueprints.todoapp.ScrollChildSwipeRefreshLayout>
+</layout>
```

Comment:
internal class LibraryImpl : Library {

    override fun addNewBook(book: Book): Boolean {
        return books.add(book)
    }

    override fun deleteBook(id: String): Boolean {
        if (books.isEmpty()) return false
        val deletableBook: Book? = books.firstOrNull { it.id == id }
        return deletableBook?.let { books.remove(it) } ?: false
    }

    override fun listBooksByName(searchString: String): Collection<String> {
        if (books.isEmpty() && searchString.isEmpty()) return emptyList()
        val bookNames: List<String> = books.map { it.name }

        val nameAndAuthorList: List<Pair<String, String>> = books.map { Pair(it.name, it.author) }

        return if (bookNames.contains(searchString)) bookNames.filter { it.contains(searchString) }
        else emptyList()
    }

    override fun listBooksByAuthor(searchString: String): List<String> {
        if (books.isEmpty() && searchString.isEmpty()) return emptyList()
        val authorNames: List<String> = books.map { it.author }
        val nameAndAuthorList: List<Pair<String, String>> = books.map { Pair(it.name, it.author) }
        return if (authorNames.contains(searchString)) authorNames.filter { it.contains(searchString) }.sortedBy { it }
        else emptyList()
    }
}